airflow/providers/facebook/ads/operators/ads.py

```diff
@@ -0,0 +1,111 @@
+
+    .. seealso::
+        For more information on the Facebook Ads Python SDK, take a look at the docs:
+        https://github.com/facebook/facebook-python-business-sdk
+
+    :param bucket: The GCS bucket to upload to
+    :type bucket: str
+    :param obj: GCS path to save the object. Must be the full file path (ex. `path/to/file.txt`)
+    :type obj: str
+    :param gcp_conn_id: Airflow Google Cloud Platform connection ID
+    :type gcp_conn_id: str
+    :param facebook_ads_conn_id: Airflow Facebook Ads connection ID
+    :type facebook_ads_conn_id: str
+    :param api_version: The version of Facebook API. Default to v6.0
+    :type api_version: str
+    :param fields: List of fields that is obtained from Facebook. Found in AdsInsights.Field class.
+        https://developers.facebook.com/docs/marketing-api/insights/parameters/v6.0
+    :type fields: List[str]
+    :param params: Parameters that determine the query for Facebook
+        https://developers.facebook.com/docs/marketing-api/insights/parameters/v6.0
+    :type params: Dict[str, Any]
+    :param gzip: Option to compress local file or file data for upload
+    :type gzip: bool
+    """
+
+    template_fields = ("facebook_ads_conn_id", "bucket", "obj")
+
+    @apply_defaults
+    def __init__(
+        self,
+        bucket: str,
+        obj: str,
+        gcp_conn_id: str = "google_cloud_default",
+        facebook_ads_conn_id: str = "facebook_ads_default",
+        api_version: str = "v6.0",
+        fields: Optional[List[str]] = None,
+        params: Optional[Dict[str, Any]] = None,
+        gzip: bool = False,
+        *args,
+        **kwargs,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.bucket = bucket
+        self.obj = obj
+        self.gcp_conn_id = gcp_conn_id
+        self.facebook_ads_conn_id = facebook_ads_conn_id
+        self.api_version = api_version
+        self.fields = fields
+        self.params = params  # type:ignore
+        self.gzip = gzip
+
+    def execute(self, context: Dict):
+        service = FacebookAdsReportingHook(facebook_ads_conn_id=self.facebook_ads_conn_id,
+                                           api_version=self.api_version)
+        rows = service.bulk_facebook_report(params=self.params, fields=self.fields)
+        try:
+            converted_rows = [dict(row) for row in rows]
+        except Exception as e:
+            self.log.error("An error occurred in converting the Facebook Ad Rows. \n Error %s", e)
+            raise
+
+        with tempfile.NamedTemporaryFile("w", suffix=".json") as json_file:
+            json.dump(converted_rows, json_file)
+            json_file.flush()
+            hook = GCSHook(gcp_conn_id=self.gcp_conn_id)
+            hook.upload(
+                bucket_name=self.bucket,
+                object_name=self.obj,
+                filename=json_file.name,
+                gzip=self.gzip,
+            )
+            self.log.info("%s uploaded to GCS", json_file.name)
```

Comment:
Should we consider return URI to the uploaded file for example `return f"gs://{self.bucket}/{self.obj}"?

Reply:
@turbaszek Looks like the gsc hook accepts either filename->"Path to that file" or it excepts data.

Reply:
True, my idea is to add a line `return f"gs://{self.bucket}/{self.obj}"` so we can ccess this value using XCom. This will help in building DAGs :)